dark/aligners.py

```diff
@@ -0,0 +1,87 @@
+from os.path import join
+from subprocess import CalledProcessError
+from tempfile import mkdtemp
+from shutil import rmtree
+
+from dark.fasta import FastaReads
+from dark.process import Executor
+from dark.reads import Reads
+
+
+def mafft(reads, verbose=False, options=None, threads=None):
+    """
+    Run a MAFT alignment and return the two sequences.
+
+    @param reads: An iterable of two reads.
+    @param verbose: If C{True} print progress info to sys.stderr.
+    @param options: A C{str} of options to pass to mafft.
+    @return: A C{Reads} instance with the two aligned sequences.
+    """
+    tempdir = mkdtemp()
+
+    infile = join(tempdir, 'input.fasta')
+    out = join(tempdir, 'result.fasta')
+
+    Reads(reads).save(infile)
+
+    if verbose:
+        print('Running mafft.', file=sys.stderr)
+
+    Executor().execute("mafft --thread %d %s '%s' > '%s'" % (
+        threads, options or '', infile, out))
```

Comment:
This will break if no value of `threads` is passed (since `None` is the default), which is my fault. Could you change it as follows:

```suggestion
    Executor().execute("mafft  %s %s '%s' > '%s'" % (
        ('' if threads is None else '--thread %d' % threads), options or '', infile, out))
```

That way we'll get the default behaviour of mafft (as opposed to us setting a default value of `1`).